packages/openneuro-server/src/datalad/snapshots.js

```diff
@@ -248,12 +249,28 @@ export const getSnapshot = (datasetId, tag) => {
       .get(url)
       .set('Accept', 'application/json')
       .then(async ({ body }) => {
-        const { created, hexsha } = await Snapshot.findOne({
+        const {
+          created,
+          hexsha,
+          deprecated,
+          deprecatedAt,
+          deprecatedBy,
+          deprecatedFor,
+        } = await Snapshot.findOne({
           datasetId,
           tag,
         }).exec()
-        const snapshot = { ...body, created, hexsha }
-        return snapshot
+        return {
+          ...body,
+          created,
+          hexsha,
+          deprecated,
+          deprecatedAt,
+          deprecatedBy: deprecatedBy
+            ? await User.findOne({ id: deprecatedBy })
+            : deprecatedBy,
```

Comment:
This can use the implicit resolver since it's just the user id, e.g. it should work if you just return the ID because of the type definition in the schema.